.github/workflows/main.yml

```
@@ -0,0 +1,37 @@
+name: Build and Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  run:
+    name: build
+    strategy:
+      fail-fast: true
+      matrix:
+        os:
+            - ubuntu-latest
+              #- macos-latest
+              #- windows-latest
+        ocaml-compiler:
+            - '4.08'
+            - '4.14'
+            - '5.1'
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@main
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+          dune-cache: true
+
+      - run: opam install -t ambient-context ambient-context-lwt ambient-context-eio --deps-only
```

this one definitely can be, right?

```suggestion
      - run: opam install . --deps-only --with-test
```

---

scratch that ❌, see below comment